src/index.js

```diff
@@ -62,7 +63,19 @@ module.exports = class{
   // setError removes errors data if an empty array or sets the errors. It also calls the changeCallback.
   setError(fieldName, errors, triggerCallback = true) {
     if(isEmpty(errors)) {
-      delete this.errors[fieldName];
+      unset(this.errors, fieldName);
+      let nested = fieldName.indexOf(".") > -1;
```

Comment:
Line 66 is all that's strictly needed to clear the error. Lines 67 - 78 are "clean up" for the errors data so that it isn't left with empty properties.